enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+
+#### Monitoring operator
+
+No change to the monitoring operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+The monitoring operator skips the alert manager's route check if `ingress` cluster operator is not there.
+
+### Risks and Mitigations
+
+#### Partially implemented capability
+
+The ingress capability will become a part of the capability API. Meanwhile its implementation will be incomplete due to the missing support of the standalone OpenShift. However the usage of the capability is an important stepping stone, it unifies the implementation of the optional ingress for all the OpenShift installation types.
+
+#### Component routes
+
+[The component routes](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#spec-componentroutes) is an optional list of routes that are managed by OpenShift components that a cluster-admin is able to configure the hostname and serving certificate for.
+The two main OpenShift components which use the component routes are:
+[the web console](https://docs.openshift.com/container-platform/4.13/web_console/customizing-the-web-console.html#customizing-the-console-route_customizing-web-console) and [the authentication](https://docs.openshift.com/container-platform/4.13/authentication/configuring-internal-oauth.html#customizing-the-oauth-server-url_configuring-internal-oauth).
+The console and the authentication operators implement the component routes by watching [the Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html) and updating the corresponding routes.    
+The ingress operator creates the required role and role binding for the OpenShift component's service account so that it can get the custom service certificate secret from the `openshift-config` namespace.  
+In the absence of the ingress operator, the component routes:
+- cannot be customized by the owner components due to the lack of the RBAC
+- become not implemented due to [the absence of the default ingress controller](#drawbacks)
+
+The cluster instance administrator will still be able use [the Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html)
+but the component owner (operator) won't be able to access the secret or rely on the route implementation.     
+This may be misleading for the cluster instance administrator. However the component routes were never fully implemented on HyperShift.
+For instance, the authentication operator which is supposed to implement the authentication component route is not used at all.
```

Comment:
The API contract for `spec.componetRoutes` states that the cluster-admin may only add an entry that has a corresponding entry in `status.componentRoutes`.  If the authentication operator and console operator do not add entries to the status, then there is nothing for the cluster-admin to configure.

Reply:
Adding the details about the API contract.